frontend/src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,40 +1,155 @@
+import { ChangeEventHandler, ReactNode, useRef } from 'react'
 import {
+  Box,
   Checkbox as ChakraCheckbox,
   CheckboxProps as ChakraCheckboxProps,
+  ComponentWithAs,
+  forwardRef,
   Icon,
+  useMergeRefs,
   useMultiStyleConfig,
 } from '@chakra-ui/react'
 
 import { BxCheckAnimated } from '~/assets/icons'
 import { CHECKBOX_THEME_KEY } from '~/theme/components/Checkbox'
 import { FieldColorScheme } from '~/theme/foundations/colours'
 
+import Input, { InputProps } from '../Input'
+
+import { CheckboxOthersContext, useCheckboxOthers } from './useCheckboxOthers'
+
 export interface CheckboxProps extends ChakraCheckboxProps {
+  /**
+   * Background and shadow colors of checkbox.
+   */
   colorScheme?: FieldColorScheme
 }
 
-export const Checkbox = ({
+type CheckboxWithOthers = ComponentWithAs<'input', CheckboxProps> & {
+  OthersCheckbox: typeof OthersCheckbox
+  OthersInput: typeof OthersInput
+  OthersWrapper: typeof OthersWrapper
+}
+
+export const Checkbox = forwardRef<CheckboxProps, 'input'>(
+  ({ children, colorScheme = 'primary', ...props }, ref) => {
+    const { icon: iconStyles } = useMultiStyleConfig(CHECKBOX_THEME_KEY, {
+      size: props.size,
+    })
+    return (
+      <ChakraCheckbox
+        icon={
+          <Icon
+            as={BxCheckAnimated}
+            __css={iconStyles}
+            // This prop needs to be passed explicitly for animations
+            isChecked={props.isChecked}
+          />
+        }
+        colorScheme={colorScheme}
+        ref={ref}
+        {...props}
+      >
+        {children}
+      </ChakraCheckbox>
+    )
+  },
+) as CheckboxWithOthers
+
+/**
+ * Components to support the "Others" option.
+ */
+
+export interface CheckboxOthersWrapperProps {
+  colorScheme?: FieldColorScheme
+  size?: string
+  children: ReactNode
+}
+
+/**
+ * Provides context values for the Others option.
+ */
+const OthersWrapper = ({
+  colorScheme,
+  size,
   children,
-  colorScheme = 'primary',
-  ...props
-}: CheckboxProps): JSX.Element => {
-  const { icon: iconStyles } = useMultiStyleConfig(CHECKBOX_THEME_KEY, {
+}: CheckboxOthersWrapperProps): JSX.Element => {
+  const checkboxRef = useRef<HTMLInputElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const styles = useMultiStyleConfig(CHECKBOX_THEME_KEY, {
+    size,
+    colorScheme,
+  })
+
+  return (
+    <CheckboxOthersContext.Provider value={{ checkboxRef, inputRef }}>
+      <Box __css={styles.othersContainer}>{children}</Box>
+    </CheckboxOthersContext.Provider>
+  )
+}
+
+/**
+ * Wrapper for the checkbox part of the Others option.
+ */
+const OthersCheckbox = forwardRef<CheckboxProps, 'input'>((props, ref) => {
+  const { checkboxRef, inputRef } = useCheckboxOthers()
+  const styles = useMultiStyleConfig(CHECKBOX_THEME_KEY, {
```

Comment:
nit: Should we just pass in `props` entirely for cleanliness?

Reply:
hmm sure, I'll add a comment there to indicate which keys are actually important though